### ISSUE TITLE
Does TabNet require feature scaling? (standardization/normalization)?

### ISSUE THREAD

**Issue by jukiewiczm** [dreamquark-ai/tabnet#356]
Hi,
Excuse me if GitHub is not the right place to ask these kind of questions - if that's the case I'm happy to ask in a more appropriate place.

The question is: Does TabNet require feature scaling as a preprocessing step before training, just like most of the NNs do? I haven't seen that being done in any of the examples - is that because of the internal batch normalization layers? What would be your recommendation?

**Comment by eduardocarvp** [comment]
Hi @jukiewiczm,

Exaclty, if you look closely at the paper and our implementation the first step is a batch normalization usually using a rather large batch size.
![image](https://user-images.githubusercontent.com/48187985/150966560-2cd6844d-2c7f-4d81-a89c-57c551d3ea59.png)
They say something to that effect in the paper as well.

**Comment by eduardocarvp** [comment]
This is where it happens in the code:

https://github.com/dreamquark-ai/tabnet/blob/95e0e9ac9637c8ed9c449d982f81d674f2016944/pytorch_tabnet/tab_network.py#L150

And quote from the paper on page 3:

"We use the raw numerical features and consider mapping of categorical features with trainable embeddings. We do not consider any global feature normalization, but merely apply batch normalization (BN)."

**Comment by jukiewiczm** [comment]
@eduardocarvp 
That's a perfect and comprehensive answer that explains it all :). Thank you very much!

**Comment by meechos** [comment]
If I may add, I have tried feature scaling and normalisation with a variety of techniques and tabnet performed consistently better without any.
